examples/token-classification/tasks.py

```diff
@@ -1,78 +1,66 @@
-                    splits = line.split(" ")
-                    words.append(splits[0])
-                    if len(splits) > 1:
-                        labels.append(splits[self.label_idx].replace("\n", ""))
-                    else:
-                        # Examples could have no label for mode = "test"
-                        labels.append("O")
-            if words:
-                examples.append(InputExample(guid=f"{mode}-{guid_index}", words=words, labels=labels))
-        return examples
-
-    def write_predictions_to_file(self, writer: TextIO, test_input_reader: TextIO, preds_list: List):
-        example_id = 0
-        for line in test_input_reader:
-            if line.startswith("-DOCSTART-") or line == "" or line == "\n":
-                writer.write(line)
-                if not preds_list[example_id]:
-                    example_id += 1
-            elif preds_list[example_id]:
-                output_line = line.split()[0] + " " + preds_list[example_id].pop(0) + "\n"
-                writer.write(output_line)
-            else:
-                logger.warning("Maximum sequence length exceeded: No prediction for '%s'.", line.split()[0])
-
-    def get_labels(self, path: str) -> List[str]:
+    def __init__(self):
+        super().__init__(source_column="tokens", target_column="labels")
+
+    def get_dataset(self, split: nlp.Split) -> nlp.Dataset:
+        return nlp.load_dataset("germeval_14", split=split)
+
+    def get_labels(self, path: Optional[str] = None) -> List[str]:
         if path:
             with open(path, "r") as f:
                 labels = f.read().splitlines()
             if "O" not in labels:
                 labels = ["O"] + labels
             return labels
         else:
-            return ["O", "B-MISC", "I-MISC", "B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC"]
+            return [
+                "B-LOC",
+                "B-LOCderiv",
+                "B-LOCpart",
+                "B-ORG",
+                "B-ORGderiv",
+                "B-ORGpart",
+                "B-OTH",
+                "B-OTHderiv",
+                "B-OTHpart",
+                "B-PER",
+                "B-PERderiv",
+                "B-PERpart",
+                "I-LOC",
+                "I-LOCderiv",
+                "I-LOCpart",
+                "I-ORG",
+                "I-ORGderiv",
+                "I-ORGpart",
+                "I-OTH",
+                "I-OTHderiv",
+                "I-OTHpart",
+                "I-PER",
+                "I-PERderiv",
+                "I-PERpart",
+                "O",
+            ]
 
 
-class Chunk(NER):
+class Chunk(TokenClassificationTask):
     def __init__(self):
-        # in CONLL2003 dataset chunk column is second-to-last
-        super().__init__(label_idx=-2)
+        super().__init__(source_column="word", target_column="chunk")
+
+    def get_dataset(self, split: nlp.Split) -> nlp.Dataset:
+        return nlp.load_dataset("./conll2003.py", split=split)
```

Comment:
CoNLL 2003 is a NER task, not a chunking one. The one about chunking is CoNLL 2000. I'm also not really in favor of using non public datasets in the examples, not everybody can have access to this dataset.

Reply:
Yes, exactly, sorry badly phrased. I mean that we don't use or evaluate the CoNLL2003 dataset for doing POS tagging or chunking they are just here to be used a features :)

Neither CoNLL 2000 or 2003 are available in nlp, that's why I proposed to add at least this one directly in nlp ;)

Reply:
Ok, no problem, I'll gladly open PRs in nlp. Thanks @jplu

Reply:
Btw: WNUT-17 would be publicly available and is already integrated into `datasets`. This is for English NER.

For Chunking, CoNLL-2000 is available via:

https://www.clips.uantwerpen.be/conll2000/chunking/

That dataset could also be integrated into `datasets` :)

Reply:
Excellent @stefan-it I'll start working on the nlp PR then.